instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java

```diff
@@ -28,17 +28,18 @@ public void consumeMessageBefore(ConsumeMessageContext context) {
       return;
     }
     Context otelContext = tracer.startSpan(Context.current(), context.getMsgList());
-    context.setMqTraceContext(otelContext);
+    context.setMqTraceContext(ContextAndScope.create(otelContext));
```

Comment:
I have some recollection of this possibly hopping threads in before / after but could be mixing up with some other instrumentation. Can you check the rocketmq code to verify? I think anytime we make a scope current in this sort of pattern that doesn't follow a try/resources form, we might do best to actually link to the library's implementation code to give a pointer to the safety of the threadlocal.

I would also not `makeCurrent` in the `create` method to show the threadlocal activation and deactivation more clearly in one file at least

Reply:
Good point; I've checked that in the sources and it always calls both before() and after() listener methods from the same thread (same method, even).